vit-servicing-station-tests/src/common/startup/db.rs

```diff
@@ -0,0 +1,179 @@
+use assert_fs::{fixture::PathChild, TempDir};
+use diesel::{connection::Connection, prelude::*};
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+use std::str::FromStr;
+use thiserror::Error;
+use vit_servicing_station_lib::db::{
+    models::{api_tokens::APITokenData, proposals::Proposal},
+    schema::{api_tokens, proposals, voteplans},
+};
+
+use std::fs;
+
+use crate::common::paths::MIGRATION_DIR;
+
+pub struct DbBuilder {
+    migrations_folder: Option<PathBuf>,
+    token: Option<APITokenData>,
+    proposals: Option<Vec<Proposal>>,
+}
+
+impl DbBuilder {
+    pub fn new() -> Self {
+        let migration_dir = PathBuf::from_str(MIGRATION_DIR).unwrap();
+
+        println!("EXE: {:?}", std::env::current_exe());
+        println!("{:?}", fs::canonicalize(&migration_dir));
+
+        Self {
+            migrations_folder: Some(migration_dir),
+            token: None,
+            proposals: None,
+        }
+    }
+
+    pub fn with_token(&mut self, token: APITokenData) -> &mut Self {
+        self.token = Some(token);
+        self
+    }
+
+    pub fn with_proposals(&mut self, proposals: Vec<Proposal>) -> &mut Self {
```

Comment:
I don't know if this is the best approach for this. The builder pattern looks good. But, this does not let us modify the DB once it is built. I think some of the functionality could go in separate functions (that we can use for the builder too). But that way, will be more flexible in case we need to modifiy the DB mid test for example.